libs/es-x-content/src/main/java/org/elasticsearch/common/xcontent/support/ParameterizedXContentParser.java

```diff
@@ -53,7 +52,7 @@ public static Map<String, Object> parse(XContentParser parser, FieldParser field
             String fieldName = parser.currentName();
             // And then the value...
             token = parser.nextToken();
-            Object value = readValue(parser, SIMPLE_MAP_FACTORY, token);
+            Object value = readValue(parser, ORDERED_MAP_FACTORY, token);
```

Comment:
I think we should either use the factory also above, or just pass in `LinkedHashMap::new` here. I see no real benefit over using these constants, I'd even remove them eventually.

Reply:
Thanks, I will go with `LinkedHashMap::new`